tests/Microsoft.Identity.Test.Unit/pop/PopAuthenticationSchemeTests.cs

```diff
@@ -129,35 +128,51 @@ public async Task ValidateKeyExpirationAsync()
                 testClock.TestTime = DateTime.UtcNow;
                 var provider = PoPProviderFactory.GetOrCreateProvider(testClock);
 
-                await app.AcquireTokenForClient(TestConstants.s_scope)
+                var result = await app.AcquireTokenForClient(TestConstants.s_scope)
                     .WithProofOfPossession(popConfig)
                     .ExecuteAsync(CancellationToken.None)
                     .ConfigureAwait(false);
+                var initialToken = result.AccessToken;
 
                 var JWK = provider.CannonicalPublicKeyJwk;
-                //Advance time 7 hours. Should still be the same key
+
+                //Advance time 7 hours. Should still be the same key and token
                 testClock.TestTime = testClock.TestTime + TimeSpan.FromSeconds(60 * 60 * 7);
 
                 provider = PoPProviderFactory.GetOrCreateProvider(testClock);
-                await app.AcquireTokenForClient(TestConstants.s_scope)
+                result = await app.AcquireTokenForClient(TestConstants.s_scope)
                     .WithProofOfPossession(popConfig)
                     .ExecuteAsync(CancellationToken.None)
                     .ConfigureAwait(false);
 
+                Assert.AreEqual(GetAccessTokenFromPopToken(result.AccessToken), GetAccessTokenFromPopToken(initialToken));
                 Assert.IsTrue(JWK == provider.CannonicalPublicKeyJwk);
+                Assert.IsTrue(result.AuthenticationResultMetadata.TokenSource == TokenSource.Cache);
+
                 //Advance time 2 hours. Should be a different key
                 testClock.TestTime = testClock.TestTime + TimeSpan.FromSeconds(60 * 60 * 2);
 
                 provider = PoPProviderFactory.GetOrCreateProvider(testClock);
-                await app.AcquireTokenForClient(TestConstants.s_scope)
+                result = await app.AcquireTokenForClient(TestConstants.s_scope)
                     .WithProofOfPossession(popConfig)
                     .ExecuteAsync(CancellationToken.None)
                     .ConfigureAwait(false);
 
                 Assert.IsTrue(JWK != provider.CannonicalPublicKeyJwk);
+                Assert.IsTrue(result.AuthenticationResultMetadata.TokenSource == TokenSource.Cache);
             }
         }
 
+        private string GetAccessTokenFromPopToken(string popToken)
```

Comment:
The name of this method does not describe what it does. It extracts the modulus, or `n` parameter from the POP token. The modulus is part of the `public key` in an RSA public / private key pair, alongside the exponent `e`. In practical terms, `e` is always the same and the `n` is different each time.

Reply:
Yes i see, I am grabbing the wrong value here, it should be getting the at